File: object-oriented-programming/encapsulation/java/player_class.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

class Player
{
    private String name;
    private int score;

    // Constructor method
    public Player(String givenName)
    {
        name = givenName;
        score = 0;
    }

    public String getName()
    {
        return name;
    }

    public int getScore()
    {
        return score;
    }

    public void setScore(int newScore)
    {
        score = newScore;
    }
}


class Testing
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        // Instantiate a new player object
        Player gamePlayer = new Player("Leona");
        System.out.println("The player's name is: " + gamePlayer.getName());
        System.out.println("The player's score is: " + gamePlayer.getScore());
        gamePlayer.setScore(20);
        System.out.println("The player's score is now: " + gamePlayer.getScore());
    }


}

File: object-oriented-programming/encapsulation/python/player_class.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0

class Player:

    def __init__(self, given_name):
        self.__name = given_name
        self.__score = 0

    def get_name(self):
        return self.__name
    
    def get_score(self):
        return self.__score
    
    def set_score(self, new_score):
        self.__score = new_score
    

# This code will run if this file is executed directly
# (i.e. not called by another program)

if __name__ == '__main__':
    # Instantiate a new player object
    game_player = Player("Leona")

    player_name = game_player.get_name()
    print(f"The player's name is: {player_name}")

    player_score = game_player.get_score()
    print(f"The player's score is: {player_score}")

    game_player.set_score(20)
    player_score = game_player.get_score()
    print(f"The player's score is now: {player_score}")
